usnistgov/pfhub
# Issue634 upload form

Fix #634

## Changes

 - Improve the upload form by allowing an arbitrary number of data and media files.

 - Enable populating of the form from existing results.

 - Simplify the metadata schema.

## Issues

Staticman does not [let existing data files update](https://github.com/eduardoboucas/staticman/pull/126) so a new simulation name is created even for an update. I might address this in later versions by having a live or dead flag in the meta.yaml files and also decoupling the simulation name from the directory name and having a unique ID for the directory names instead.

 - The tooltips still don't work. That is in a different [issue](https://github.com/usnistgov/chimad-phase-field/issues/385) to be addressed soon.

## How to Review

These changes can be [viewed live][live-site].

 - Check that you like the look and feel of the [upload form](http://random-cat-668.surge.sh/simulations/upload_form/).

 - Try entering data and submitting. Name the submission (short name) "fake" so that I know it is a fake submission

 -  Try populating from an existing simulation. Go to the [simulation result table](http://random-cat-668.surge.sh/simulations/#simulations). Click on a simulation link - it should open up an individual simulation result landing page. Click on the purple edit button to the side of name of the simulation. It should populate the form. Try changing (or not) some of the data in the form and submitting.

 - Try submitting new data files and media files - anything will do, doesn't have to be real data

**Remember to tear down the [live site][live-site] when merging or
  closing this pull-request.**

[live-site]: http://random-cat-668.surge.sh


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/usnistgov/chimad-phase-field/668)
